objects/src/block/header.rs

```diff
@@ -145,12 +145,12 @@ impl BlockHeader {
     }
 
     /// Returns the protocol version.
-    pub fn version(&self) -> Felt {
+    pub fn version(&self) -> u32 {
         self.version
     }
 
     /// Returns the timestamp at which the block was created.
-    pub fn timestamp(&self) -> Felt {
+    pub fn timestamp(&self) -> u32 {
```

Comment:
Please add a comment about the unit of `timestamp`, and a rough outline of the the expected time of 136 years:

```
>>> (2 ** 32) / (60 * 60 * 24 * 365)
136.19251953323186
```

Note: I personally think this is a bad idea, saving 32bits is not worth the hassle, but that is just my opinion.

Reply:
Thank you for clarification. Storage in 32 bits is more convenient, because it requires less conversions, you can see it in the changes. Since we count time in unsigned 32 integer starting from UNIX epoch (1970), the latest timestamp will be in 2106, in ~82 years. It will be much later than "year 2038 problem" which UNIX systems and different apps/DBMS were expected to face because of signed 32-bits timestamps. We could introduce "Miden epoch" (start counting from 2024, or 2020 for conveniency) to shift this limitation to even later time, but I don't think this would be convenient for developers who use it. And in 82 years, IMO, blockchains and Miden, as well, will be completely different, at least several generation will be changed, and block format will be redesigned a lot of times and there will be a lot of different breaking changes. Thus, time storage will be also redesigned completely.